Sources/DeepLearning/Layer.swift

```diff
@@ -1404,12 +1409,75 @@ public struct LSTMCell<Scalar: TensorFlowFloatingPoint>: RNNCell, VectorNumeric
     }
 }
 
+/// A gated recurrent unit (GRU) cell.
+///
+/// Reference: [Gated recurrent unit](https://en.wikipedia.org/wiki/Gated_recurrent_unit).
+@_fixed_layout
+public struct GRUCell<Scalar: TensorFlowFloatingPoint>: RNNCell {
+    public var resetWeight, updateWeight, inputHiddenWeight, hiddenHiddenWeight: Tensor<Scalar>
+    public var resetBias, updateBias, inputHiddenBias, hiddenHiddenBias: Tensor<Scalar>
+
+    public typealias TimeStepInput = Tensor<Scalar>
+    public typealias TimeStepOutput = Tensor<Scalar>
+    public typealias State = Tensor<Scalar>
+    public typealias Input = RNNCellInput<TimeStepInput, State>
+    public typealias Output = RNNCellOutput<TimeStepOutput, State>
+
+    @noDerivative
+    public var stateShape: TensorShape {
+        return TensorShape([1, resetWeight.shape[1]])
+    }
+
+    public var zeroState: State {
+        return Tensor(zeros: stateShape)
+    }
+
+    /// Creates a `GRUCell` with the specified input size and hidden state size.
+    ///
+    /// - Parameters:
+    ///   - inputSize: The number of features in 2-D input tensors.
+    ///   - hiddenSize: The number of features in 2-D hidden states.
+    ///   - seed: The random seed for initialization. The default value is random.
+    public init(inputSize: Int, hiddenSize: Int,
+                seed: (Int64, Int64) = (Int64.random(in: Int64.min..<Int64.max),
+                                        Int64.random(in: Int64.min..<Int64.max))) {
+        let concatenatedHiddenSize = inputSize + hiddenSize
+        let weightShape: TensorShape = [concatenatedHiddenSize, hiddenSize]
+        let biasShape: TensorShape = [hiddenSize]
+        self.resetWeight = Tensor(glorotUniform: weightShape, seed: seed)
+        self.resetBias = Tensor(zeros: biasShape)
+        self.updateWeight = Tensor(glorotUniform: weightShape, seed: seed)
+        self.updateBias = Tensor(repeating: -1, shape: biasShape)
+        self.inputHiddenWeight = Tensor(glorotUniform: [inputSize, hiddenSize], seed: seed)
+        self.inputHiddenBias = Tensor(zeros: biasShape)
+        self.hiddenHiddenWeight = Tensor(glorotUniform: [hiddenSize, hiddenSize], seed: seed)
+        self.hiddenHiddenBias = Tensor(zeros: biasShape)
+    }
+
+    /// Returns the output obtained from applying the layer to the given input.
+    ///
+    /// - Parameters:
+    ///   - input: The input to the layer.
```

Comment:
Minor note (totally need not be addressed here): `- Parameter input: ` and `- Parameters: - input:` are used inconsistently in `Layer.swift`.

Reply:
Most layers are using a separate line for the `input` parameter. This is legacy left by the fact that we had a context before. Will change later.